test/e2e/network/network_policy.go

```diff
@@ -532,12 +640,601 @@ var _ = SIGDescribe("NetworkPolicy", func() {
+			protocolUDP := v1.ProtocolUDP
+
+			// Getting podServer's status to get podServer's IP, to create the CIDR
+			podServerStatus, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Get(podServer.Name, metav1.GetOptions{})
+			if err != nil {
+				framework.ExpectNoError(err, "Error occurred while getting pod status.")
+			}
+
+			podServerCIDR := fmt.Sprintf("%s/32", podServerStatus.Status.PodIP)
+
+			// Creating pod-b and service-b
+			podServerB, serviceB = createServerPodAndService(f, f.Namespace, "pod-b", []int{80})
+			ginkgo.By("Waiting for pod-b to be ready", func() {
+				err = f.WaitForPodReady(podServerB.Name)
+				framework.ExpectNoError(err, "Error occurred while waiting for pod type: Ready.")
+			})
+			defer cleanupServerPodAndService(f, podServerB, serviceB)
+
+			// Wait for podServerB with serviceB to be ready
+			err = e2epod.WaitForPodRunningInNamespace(f.ClientSet, podServerB)
+			framework.ExpectNoError(err, "Error occurred while waiting for pod status in namespace: Running.")
+
+			// Before creation of the policy client-a should be able to connect to server-b
+			ginkgo.By("Creating client-a which should be able to contact the server-b.", func() {
+				testCanConnect(f, f.Namespace, "client-a", serviceB, 80)
+			})
+
+			policyAllowCIDR := &networkingv1.NetworkPolicy{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: f.Namespace.Name,
+					Name:      "allow-client-a-via-cidr-egress-rule",
+				},
+				Spec: networkingv1.NetworkPolicySpec{
+					// Apply this policy to the Server
+					PodSelector: metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"pod-name": "client-a",
+						},
+					},
+					PolicyTypes: []networkingv1.PolicyType{networkingv1.PolicyTypeEgress},
+					// Allow traffic to only one CIDR block.
+					Egress: []networkingv1.NetworkPolicyEgressRule{
+						{
+							Ports: []networkingv1.NetworkPolicyPort{
+								// Allow DNS look-ups
+								{
+									Protocol: &protocolUDP,
+									Port:     &intstr.IntOrString{Type: intstr.Int, IntVal: 53},
+								},
+							},
+						},
+						{
+							To: []networkingv1.NetworkPolicyPeer{
+								{
+									IPBlock: &networkingv1.IPBlock{
+										CIDR: podServerCIDR,
+									},
+								},
+							},
+						},
+					},
+				},
+			}
+
+			policyAllowCIDR, err = f.ClientSet.NetworkingV1().NetworkPolicies(f.Namespace.Name).Create(policyAllowCIDR)
+			framework.ExpectNoError(err, "Error occurred while creating policy: policyAllowCIDR.")
+			defer cleanupNetworkPolicy(f, policyAllowCIDR)
+
+			ginkgo.By("Creating client-a which should not be able to contact the server-b.", func() {
+				testCannotConnect(f, f.Namespace, "client-a", serviceB, 80)
+			})
+
+			ginkgo.By("Creating client-a which should be able to contact the server.", func() {
+				testCanConnect(f, f.Namespace, "client-a", service, 80)
+			})
+		})
+	})
+
+	ginkgo.Context("NetworkPolicy between pods, each containing a server container and a client container", func() {
```

Comment:
I'm still dubious about this test, and there is a push on to use fewer images and fewer test programs (eg, to make tests work on Windows pods, which this test case certainly wouldn't). So I think you should just remove this for now. (And remove `createPodWithClientServerAndService` along with it.)

Reply:
I have changed the implementation of this test. I have removed the `createPodWithClientServerAndService `.